editoast/src/views/timetable.rs

```diff
@@ -451,13 +451,28 @@ impl Conflict {
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
 struct PacedTrainOccurrenceId {
     paced_train_id: i64,
-    index: u64,
+    #[schema(inline)]
+    #[serde(flatten)]
+    occurrence_ref: PacedTrainOccurrenceRef,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
+#[serde(untagged)]
+enum PacedTrainOccurrenceRef {
+    BaseOccurrence { index: u64 },
+    ModifiedException { index: u64, exception_key: String },
+    CreatedException { exception_key: String },
+}
+
+#[derive(Debug, Clone, PartialEq)]
+/// This ID is used to identify paced train occurrences and exceptions when sending them to the core API for conflict detection.
 enum TrainId {
+    /// Id format `{id}`
     TrainSchedule(i64),
-    PacedTrainOccurrence { paced_train_id: i64, index: u64 },
+    PacedTrainOccurrence {
+        paced_train_id: i64,
+        occurrence_ref: PacedTrainOccurrenceRef,
+    },
```

Comment:
`TrainId` is an internal object, whereas `PacedTrainOccurrenceRef` is meant to be an API schema. Could you replace this variant by these three `PacedTrainBaseOccurence`, `PacedTrainModifiedOccurrence`, `PacedTrainCreatedOccurence`. That'll also make the `FromStr` core more explicit.

Reply:
Is it not recommended to mix schema structs with internal structs? Could you please clarify the reasons behind this change, and how it makes the FromStr implementation clearer ?